tools/src/addr-convert/Main.hs

```diff
@@ -50,27 +59,54 @@ usageExample :: Maybe Doc
 usageExample = (Just . fromString @Doc . toString @Text) [text|
 Command example:
 
-  stack exec -- cardano-addr-convert -a 2HF83bvYCTzoCbVta6t64W8rFEnvnkJbIUFoT5tOyoU=
+  stack exec -- cardano-addr-convert -a 2HF83bvYCTzoCbVta6t64W8rFEnvnkJbIUFoT5tOyoU= -p mainnet
 
 Output example:
 
-  3mhNKjfhaCT13DjcQ9eMK4VHfZrFxmyXq8SjVPRtz7SWfP
+  Ae2tdPwUPEZLTt73QktZUahV9FXnBRJNgzpbueBqQKGyPDgPdzPdwVretyk
 
-You can also run it without arguments to switch to interactive mode.
+You can also run it without the address argument to switch to interactive mode.
 In this case each entered vending address is echoed with a testnet address.|]
 
-convertAddr :: Text -> IO Text
-convertAddr addr =
-    pretty . makeRedeemAddress fixedNM <$>
+convertAddr :: NetworkMagic -> Text -> IO Text
+convertAddr nm addr =
+    pretty . (makeRedeemAddress nm) <$>
     (eitherToThrow . fromAvvmPk) (toText addr)
 
+-- Both mainnet & staging use NetworkMainOrStage. We explicity disallow provision of
+-- their integer values, so as to avoid confusion with NetworkTestnet.
+toNetworkMagic :: Text -> Either Text NetworkMagic
+toNetworkMagic txt =
+    case txt of
+        "mainnet" -> Right NetworkMainOrStage
+        "staging" -> Right NetworkMainOrStage
+
+        num -> case decimal num of
+            Right (pm, _)
+                | pm == stagingProtocolMagic ->
+                    Left "Got staging's ProtocolMagic;\
+                        \ please enter 'staging' instead"
+                | pm == mainnetProtocolMagic ->
+                    Left "Got mainnet's ProtocolMagic;\
+                        \ please enter 'mainnet' instead"
+            Right (ident, _) ->
+                Right (makeNetworkMagic (ProtocolMagic (ProtocolMagicId ident)
+                                        RequiresMagic))
+            Left err -> Left $ toText ("Please enter either 'mainnet', 'staging', or\
+                                      \ an Int32 value: " ++ err)
```

Comment:
Ok, so IMO the help text in the CLI parser should have a message like:

`Valid values include 'mainnet', 'staging', or an Int32 value to indicate a test network.`

Reply:
Done!